modules/tinymce/src/core/test/ts/browser/annotate/AnnotationStylingTest.ts

```diff
@@ -0,0 +1,311 @@
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    setup: (editor: Editor) => {
+      editor.on('init', () => {
+        editor.annotator.register('test-annotation', {
+          decorate: (_uid, _data) => ({
+            classes: []
+          })
+        });
+        editor.annotator.register('test-comment', {
+          decorate: (_uid, _data) => ({
+            classes: [ 'tox-comment' ]
+          })
+        });
+      });
+    }
+  }, [], true);
+
+  const noOutline: Outline = {
+    color: 'rgb(0, 0, 0)',
+    width: '0px',
+    style: 'none'
+  };
+
+  const emptyFigCaptionOutline: Outline = {
+    color: 'rgb(153, 153, 153)',
+    width: '0px',
+    style: 'none'
+  };
+
+  const selectedOutline: Outline = {
+    color: 'rgb(180, 215, 255)', // #b4d7ff
+    width: '3px',
+    style: 'solid'
+  };
+
+  const commentOutline: Outline = {
+    color: 'rgb(255, 232, 157)', // #ffe89d
+    width: '3px',
+    style: 'solid'
+  };
+
+  const commentActiveOutline: Outline = {
+    color: 'rgb(254, 214, 53)', // #fed635
+    width: '3px',
+    style: 'solid'
+  };
+
+  const noBackgroundColor = 'rgba(0, 0, 0, 0)';
+  const commentBackgroundColor = 'rgb(255, 232, 157)'; // #ffe89d
+  const commentActiveBackgroundColor = 'rgb(254, 214, 53)'; // #fed635
+  const inlineBoundaryBackgroundColor = 'rgb(180, 215, 255)'; // #b4d7ff
+
+  const getOutline = (elm: SugarElement<Element>): Outline => {
+    const color = Css.get(elm, 'outline-color');
+    const width = Css.get(elm, 'outline-width');
+    const style = Css.get(elm, 'outline-style');
+    return {
+      color,
+      width,
+      style
+    };
+  };
+
+  const getBackgroundColor = (elm: SugarElement<Element>) =>
+    Css.get(elm, 'background-color');
+
+  const pAssertStyling = (editor: Editor, selector: string, expectedBackgroundColor: string, expectedOutline: Outline, checkOtherNodes: boolean = true) =>
+    Waiter.pTryUntil('Should have correct styling', () => {
+      const body = TinyDom.body(editor);
+      const elm = UiFinder.findIn(body, selector).getOrDie();
+      const actualBackgroundColor = getBackgroundColor(elm);
+      const actualOutline = getOutline(elm);
+      assert.equal(actualBackgroundColor, expectedBackgroundColor);
+      assert.deepEqual(actualOutline, expectedOutline);
+
+      if (checkOtherNodes) {
+        const parents = SelectorFilter.ancestors(elm, '*', (e) => Compare.eq(e, body));
+        const children = SelectorFilter.children(elm, '*');
+        const isFigCaption = SugarNode.isTag('figcaption');
+
+        Arr.each(parents, (e) => assert.deepEqual(getOutline(e), noOutline, 'parent should not have outline'));
+        Arr.each(children, (e) => assert.deepEqual(getOutline(e), isFigCaption(e) ? emptyFigCaptionOutline : noOutline, 'child should not have outline'));
+      }
+    });
+
+  before(() => {
+    const editor = hook.editor();
+    Class.add(TinyDom.body(editor), 'tox-comments-visible');
+  });
```

Comment:
This should be up the top near the other hooks to keep everything grouped (e.g. `TinyHook.bddSetup`)